src/runtime_src/core/edge/drm/zocl/zocl_aie.c

```diff
@@ -0,0 +1,102 @@
+/* SPDX-License-Identifier: GPL-2.0 OR Apache-2.0 */
+/*
+ * A GEM style (optionally CMA backed) device manager for ZynQ based
+ * OpenCL accelerators.
+ *
+ * Copyright (C) 2020 Xilinx, Inc. All rights reserved.
+ *
+ * Author(s): Himanshu Choudhary <hchoudha@xilinx.com>
+ *
+ * This file is dual-licensed; you may select either the GNU General Public
+ * License version 2 or Apache License, Version 2.0.
+ */
+
+#include "ert.h"
+#include "zocl_drv.h"
+#include "zocl_aie.h"
+#include "sched_exec.h"
+
+int
+zocl_aie_getcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
+{
+	struct drm_zocl_dev *zdev = dev->dev_private;
+	struct aie_info *aie = zdev->aie_information;
+	struct aie_info_cmd *acmd;
+	struct drm_zocl_aie_cmd *kdata = data;
+
+	// If no command, the process who calls this ioctl will block here
+	mutex_lock(&aie->aie_lock);
+	while (list_empty(&aie->aie_cmd_list)) {
+		mutex_unlock(&aie->aie_lock);
+		if (wait_event_interruptible(aie->aie_wait_queue,
+		    !list_empty(&aie->aie_cmd_list))) {
+			return -ERESTARTSYS;
+                }
+		mutex_lock(&aie->aie_lock);
+	}
+
+	acmd = list_first_entry(&aie->aie_cmd_list, struct aie_info_cmd,
+	    aiec_list);
+	list_del(&acmd->aiec_list);
+	mutex_unlock(&aie->aie_lock);
+
+	kdata->opcode = acmd->aiec_packet->opcode;
+	kfree(acmd->aiec_packet);
+	kfree(acmd);
+	return 0;
+}
+int
+zocl_aie_putcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
+{
+	struct drm_zocl_dev *zdev = dev->dev_private;
+	struct aie_info *aie = zdev->aie_information;
+	struct aie_info_cmd *acmd;
+	struct drm_zocl_aie_cmd *kdata = data;
+	struct aie_info_packet *aiec_packet;
+
+	acmd = kmalloc(sizeof(struct aie_info_cmd), GFP_KERNEL);
+	if (!acmd) {
+		return -ENOMEM;
+	}
+
+	aiec_packet = kmalloc(sizeof(struct aie_info_packet), GFP_KERNEL);
+	if (!aiec_packet) {
+		return -ENOMEM;
+	}
+
+	aiec_packet->opcode = kdata->opcode;
+	acmd->aiec_packet = aiec_packet;
+	struct aie_info_packet *cmd;
+	cmd = acmd->aiec_packet;
+	cmd->size = kdata->size;
+	snprintf(cmd->info, (cmd->size < AIE_INFO_SIZE) ? cmd->size:AIE_INFO_SIZE
+		, "%s", (char *)kdata->info);
```

Comment:
it is better to put "," right after AIE_INFO_SIZE in the same line

Reply:
done